cmake/SearchForStuff.cmake

```diff
@@ -126,6 +126,13 @@ macro (check_gcc_visibility)
   check_cxx_compiler_flag(-fvisibility=hidden GCC_SUPPORTS_VISIBILITY)
 endmacro()
 
+########################################
+# Find ignition cmake2
+# Only for using the testing macros and creating the codecheck target, not 
+# really being use to configure the whole project
+find_package(ignition-cmake2 2.3 REQUIRED)
```

Comment:
Is there a particular reason why version `2.3` is the minimum required version? It looks like `ign_setup_target_for_codecheck` exists before `2.3`:  https://github.com/ignitionrobotics/ign-cmake/blob/ignition-cmake2_2.2.0/cmake/IgnCodeCheck.cmake#L1-L2

Reply:
This will actually need to be updated since this PR needs a new release of ign-cmake to ensure `make codecheck` is clean.

Reply:
I wouldn't add that as a version requirement, since that will cause build failures if people have an outdated version of ign-cmake, which is a bit extreme to ensure correct linting. I would recommend a `find_package` call with no version requirement, since `IgnCodeCheck` was present in ign-cmake's 2.0.0 release

Reply:
That's a good point; I think I agree with @scpeters comments.

Reply:
Makes sense. If it won't cause failure in CI, since we assume CI will have the latest ign-cmake, we can remove the version requirement.

Also, I originally thought https://github.com/ignitionrobotics/ign-cmake/pull/187 would be needed to make CI pass, but it turns out without that PR, we do get output from codechech, but they are just warnings, not errors. So technically, this PR can go in without the ign-cmake release.

Reply:
https://github.com/ignitionrobotics/sdformat/pull/682/commits/9a6d507226b7ccba33011b4a59043a28499ad03a